examples/step-21/step-21.cc

```diff
@@ -884,7 +889,7 @@ namespace Step21
 
                 for (unsigned int i = 0; i < dofs_per_cell; ++i)
                   local_rhs(i) -=
-                    time_step * normal_flux *
+                    time.get_next_step_size() * normal_flux *
```

Comment:
Its late at night and this is bending my brain a bit, so I hope I hadn't missed something elementary when I write this.

I think that this should be `time.get_previous_step_size()`, given the current definition of the function. I believe that here, and typically*, one is solving for the solution at the current time, so `delta_t = current_time - previous_time`. Furthermore, if `current_time == end_time` (which, due to the guard in the driving timestepping loop, it cannot be. However, it does not preclude someone from not using the `time.is_at_end()` function...), then `next_step_size` would either be `0`, or undefined.

(*) I see though that this tutorial has some initial conditions, but then proceeds to find the solution at `t=0`. @bangerth Is this perhaps a mistake? Aren't initial conditions those defined at `t=0`, or at some reasonable `t=t_{0}`? The setup infers that `t_{0} = -time_step`, which seems weird to me. In step-44, we specifically laid out the initial conditions at `t=0` and immediately incremented time afterwards, just to avoid such an ambiguity.

Reply:
Correction: Step-21 is an explicit method, and it solves for the solution at time `current_time + next_step_size`. So this is indeed correct.